modules/javafx.base/src/test/java/test/javafx/util/converter/LocalDateStringConverterTest.java

```diff
@@ -25,144 +25,116 @@
 
+import static org.junit.jupiter.api.Assertions.*;
+
 import java.time.LocalDate;
 import java.time.chrono.IsoChronology;
 import java.time.format.DateTimeFormatter;
 import java.time.format.FormatStyle;
 import java.util.Locale;
 import java.util.stream.Stream;
 
-import javafx.util.converter.LocalDateStringConverter;
-import javafx.util.converter.LocalDateStringConverterShim;
-
-import static org.junit.jupiter.api.Assertions.*;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
-public class LocalDateStringConverterTest {
-    private static final LocalDate VALID_DATE = LocalDate.of(1985, 1, 12);
+import javafx.util.converter.BaseTemporalStringConverterShim;
+import javafx.util.converter.LocalDateStringConverter;
 
-    private static Locale oldLocale = null;
-    private static DateTimeFormatter aFormatter = null;
-    private static DateTimeFormatter aParser = null;
+public class LocalDateStringConverterTest {
 
-    private enum LocalDateStringConverterVariant {
-        NO_PARAM,
-        WITH_FORMATTER_PARSER,
-        WITH_FORMAT_STYLES,
-    }
+    private static final LocalDate VALID_DATE = LocalDate.of(1985, 1, 12);
 
-    static Stream<Arguments> provideTestParameters() {
-        return Stream.of(
-                Arguments.of(LocalDateStringConverterVariant.NO_PARAM, FormatStyle.SHORT, VALID_DATE),
-                Arguments.of(LocalDateStringConverterVariant.WITH_FORMATTER_PARSER, null, VALID_DATE),
-                Arguments.of(LocalDateStringConverterVariant.WITH_FORMAT_STYLES, FormatStyle.SHORT, VALID_DATE)
-        );
-    }
+    private static Locale oldLocale;
+    private static DateTimeFormatter aFormatter;
+    private static DateTimeFormatter aParser;
 
     @BeforeAll
     public static void setupBeforeAll() {
         oldLocale = Locale.getDefault();
+        // Tests require that default locale is en_US
         Locale.setDefault(Locale.US);
-
+        // DateTimeFormatter uses default locale, so we can init this after updating locale
         aFormatter = DateTimeFormatter.ofPattern("dd MM yyyy");
         aParser = DateTimeFormatter.ofPattern("yyyy MM dd");
     }
 
     @AfterAll
     public static void teardownAfterAll() {
+        // Restore VM's old locale
         Locale.setDefault(oldLocale);
     }
 
-    private record ConverterSetup(
-            LocalDateStringConverter converter,
-            Locale locale,
-            DateTimeFormatter formatter,
-            DateTimeFormatter parser
-    ) {}
-
-    private ConverterSetup setupConverter(
-            LocalDateStringConverterVariant converterVariant,
-            FormatStyle dateStyle,
-            LocalDate validDate
-    ) {
+    private enum LocalDateStringConverterVariant {
+        NO_PARAM,
+        WITH_FORMATTER_PARSER,
+        WITH_FORMAT_STYLES,
```

Comment:
I know enums allow it, but trailing comma is still ugly

Reply:
I think that in JavaFX we use a `;`. The comma stayed from the old code, but I fixed it.